packages/flutter_tools/lib/src/isolated/resident_web_runner.dart

```diff
@@ -812,86 +821,100 @@ Please provide a valid TCP port (an integer between 0 and 65535, inclusive).
     Uri? websocketUri;
     if (supportsServiceProtocol) {
       assert(connectDebug != null);
-      _connectionResult = await connectDebug;
-      unawaited(_connectionResult!.debugConnection!.onDone.whenComplete(_cleanupAndExit));
+      unawaited(
```

Comment:
The diff is making it tricky to see but is this the key difference that is breaking the deadlock we were discussing?

Reply:
That's correct. The `app.started` event needs to go out for the tool to start the test driver process, and the VM service doesn't need to be configured before that's sent out since there's a separate `app.debugPort` event used to indicated that the service is available.

Attaching a listener to `connectDebug` instead of blocking on its completion and not requiring the service to be available before application startup lets us break the deadlock safely.